src/api.rs

```diff
@@ -976,9 +976,18 @@ impl<T: Pixel> ContextInner<T> {
             == 0
             && prev_input_frameno == (next_keyframe_input_frameno - 1))
         {
+          input_frameno = next_keyframe_input_frameno;
+
+          // If we'll return early, do it before modifying the state.
+          match self.frame_q.get(&input_frameno) {
```

Comment:
frame_q.contains() ?

Reply:
Flush is recorded in this queue as `None`, and we're not interested in the flush, so we need `Some(Some(_))`.

Reply:
ok